chemprop/utils.py

```diff
@@ -768,3 +770,29 @@ def update_prediction_args(
                 "(with either --features_generator or --features_path "
                 "and using --no_features_scaling if applicable)."
             )
+
+
+def multitask_mean(
+    scores: np.ndarray,
+    metric: str,
+    axis: int = None,
+) -> float:
+    """
+    A function for combining the metric scores across different
+    model tasks into a single score. When the metric being used
+    is one that varies with the magnitude of the task (such as RMSE),
+    a geometric mean is used, otherwise a more typical arithmetic mean
+    is used. This prevents a task with a larger magnitude from dominating
+    over one with a smaller magnitude (e.g., temperature and pressure).
+
+    :param scores: The scores from different tasks for a single metric.
+    :param metric: The metric used to generate the scores.
+    :axis: The axis along which to take the mean.
+    :return: The combined score across the tasks.
+    """
+    scale_dependent_metrics = ['rmse', 'mae', 'mse', 'bounded_rmse', 'bounded_mae', 'bounded_mse']
+
+    if metric in scale_dependent_metrics:
+        return gmean(scores, axis=axis)
+    else:
```

Comment:
Might I suggest explicitly including the non-scale dependent metrics here, and raise an error if the metric isn't in either the scale-dependent or non-scale-dependent lists? That way, in a later PR, if someone wants to add a new scale-dependent metric but forgets to update this function, then they get an informative error message.

Reply:
This is a good note. I’ll add something like this.

Reply:
Okay, added. Let me know what you think.

Reply:
Looks good!